workspaces/rbac/plugins/rbac-backend/src/service/permission-definition-routes.ts

```diff
@@ -0,0 +1,171 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  PermissionDependentPluginList,
+  policyEntityCreatePermission,
+  policyEntityDeletePermission,
+  policyEntityReadPermission,
+} from '@backstage-community/plugin-rbac-common';
+import Router from 'express-promise-router';
+import { logAuditorEvent, setAuditorError } from '../auditor/rest-interceptor';
+import { PluginPermissionMetadataCollector } from './plugin-endpoints';
+import {
+  PermissionDependentPluginDTO,
+  PermissionDependentPluginStore,
+} from '../database/extra-permission-enabled-plugins-storage';
+import { authorizeConditional } from './policies-rest-api';
+import {
+  AuditorService,
+  AuthService,
+  HttpAuthService,
+  PermissionsService,
+} from '@backstage/backend-plugin-api';
+import { ExtendablePluginIdProvider } from './extendable-id-provider';
+import {
+  ConflictError,
+  NotAllowedError,
+  NotFoundError,
+} from '@backstage/errors';
+
+// todo: do we need separated permissions to set up extra plugin ids  ?
+export function createPermissionDefinitionRoutes(
+  pluginPermMetaData: PluginPermissionMetadataCollector,
+  pluginIdProvider: ExtendablePluginIdProvider,
+  extraPluginsIdStorage: PermissionDependentPluginStore,
+  deps: {
+    auth: AuthService;
+    httpAuth: HttpAuthService;
+    auditor: AuditorService;
+    permissions: PermissionsService;
+  },
+) {
+  const router = Router();
+
+  const { auth, auditor } = deps;
+
+  router.get(
+    '/plugins/policies',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityReadPermission, deps);
+
+      const body = await pluginPermMetaData.getPluginPolicies(auth);
+
+      response.json(body);
+    },
+  );
+
+  router.get(
+    '/plugins/condition-rules',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityReadPermission, deps);
+
+      const body = await pluginPermMetaData.getPluginConditionRules(auth);
+
+      response.json(body);
+    },
+  );
+
+  router.get(
+    '/plugins/id',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityReadPermission, deps);
+
+      const actualPluginIds = await pluginIdProvider.getPluginIds();
+      response.status(200).json(pluginIdsToResponse(actualPluginIds));
+    },
+  );
+
+  router.post(
+    '/plugins/id',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityCreatePermission, deps);
+      const pluginIds: PermissionDependentPluginList = request.body;
+      // todo validate pluginIds object
```

Comment:
Were you still planning on completing this todo?

Reply:
Done.